### PR TITLE
allocator: Add depth to format table

### PR DESCRIPTION
**Pull request by Drakulix**

### CODE REVIEW

File: src/backend/allocator/format.rs [hunk 1]
```diff
@@ -82,6 +83,18 @@ macro_rules! format_tables {
             }
         }
 
+        /// Returns the depth of the specified format.
```
Comment (i509VCB) [inline src/backend/allocator/format.rs:86]: what makes a depth different from the bpp?

Comment (i509VCB) [reply src/backend/allocator/format.rs:86]: Also let's mention this in the module docs

File: src/backend/allocator/format.rs [hunk 2]
```diff
@@ -94,218 +107,238 @@ macro_rules! format_tables {
 
 format_tables! {
     // 8-bit bpp Red
-    R8 { alpha: false, bpp: 8 },
+    R8 { alpha: false, bpp: 8, depth: 8 },
 
     // TODO: Update drm-fourcc
     // 16-bit bpp Red with padding (x:R)
     // R10 { bpp: 16 },
     // R12 { bpp: 16 },
```
Comment (i509VCB) [inline src/backend/allocator/format.rs:115]: Let's add the depth to these also, we do need to update drm-fourcc to get these variants though

File: src/backend/allocator/format.rs (hunk 1 above)
Comment (Drakulix) [reply src/backend/allocator/format.rs:86]: It is the amount of used bits per pixel. So any padding or non-alpha ("X") parts of the format are not counted towards the depth, but the bpp.

Older drm apis require that information to create a framebuffer.

Comment (Drakulix) [reply src/backend/allocator/format.rs:96]: Added to module docs

File: src/backend/allocator/format.rs (hunk 2 above)
Comment (Drakulix) [reply src/backend/allocator/format.rs]: done

File: src/backend/allocator/format.rs [hunk 3]
```diff
@@ -26,6 +26,16 @@
...
 //! assert_eq!(get_bpp(Fourcc::Argb8888), Some(32));
 //! ```
+//!
+//! [`get_depth`] returns the number of used bits per pixel of a format
+//! (excluding padding or non-alpha "X" parts of the format).
+//!
+//! ```
+//! # use smithay::backend::allocator::Fourcc;
+//! # use smithay::backend::allocator::format::get_depth;
+//! assert_eq!(get_depth(Fourcc::Argb8888), Some(32))
```
Comment (i509VCB) [inline src/backend/allocator/format.rs:36]: doc tests failed

**Review by vberger — APPROVED** [review]

**Pull request merged**